src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -1,23 +1,50 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 /**
  * Реалізуйте методи для знаходження головної та побічної діагоналі квадратної матриці,
  * максимального та мінімального елементів.
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] result = new int[matrix[0].length];
+        for (int i = 0; i < result.length; i++) {
+            result[i] = matrix[i][i];
+        }
+        return result;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        String s = Arrays.deepToString(matrix);
+        int[] result = new int[matrix[0].length];
+        for (int i = 0; i < result.length; i++) {
+            result[i] = matrix[i][matrix[0].length - i - 1];
```

Comment:
```suggestion
            result[i] = matrix[i][matrix[i].length - i - 1];
```
can we avoid hardcoding here?